src/utils/component.js

```diff
@@ -39,4 +44,173 @@ const traverseFragment = (component) => {
   return fragmentComponents;
 };
 
-export { traverseFragment, findReactRootContainer };
+const colorize = (text, styles) => {
+  return [`%c${text}`, styles];
+};
+
+const findReactRootNode = (element) => {
+  if (!element) return null;
+
+  if (
+    Object.keys(element).some(
+      (key) =>
+        key.startsWith('__reactFiber$') ||
+        key.startsWith('__reactInternalInstance$'),
+    )
+  ) {
+    return element;
+  }
+
+  const childrenArray = Array.from(element.children);
+  let reactNode = null;
+
+  childrenArray.some((children) => {
+    reactNode = findReactRootNode(children);
+
+    return reactNode;
+  });
+
+  return reactNode;
+};
+
+const findReactFiber = (element) => {
+  const reactKey = Object.keys(element).find(
+    (key) =>
+      key.startsWith('__reactFiber$') ||
+      key.startsWith('__reactInternalInstance$'),
+  );
+
+  return reactKey ? element[reactKey] : null;
+};
+
+const findHostComponent = (fiber) => {
+  let node = fiber;
+
+  while (node) {
+    if (node.stateNode instanceof HTMLElement) {
+      return node.stateNode;
+    }
+
+    node = node.child;
+  }
+
+  return null;
+};
+
+const traverseFiberTree = (
+  fiber,
+  depth = 0,
+  isLast = true,
+  prefix = '',
+  isRoot = false,
+) => {
+  if (!fiber) return;
+
+  const connector = isLast ? '└─' : '├─';
+  const line = depth > 0 ? `${prefix}${connector}` : '';
+
+  if (isRoot) {
+    const domElement = findHostComponent(fiber);
+    const componentType = '▸';
+
+    const [styledComponentName, styleString] = colorize(
+      'App',
+      ROOT_COMPONENT_BLOCK_STYLE,
+    );
+
+    console.log(
+      `${line}${componentType} ${styledComponentName}`,
+      styleString,
+      domElement,
+    );
```

Comment:
traverseFiberTree 함수 내에 중복된 코드가 있는 것 같습니다! 
위 코드를 함수로 분리해서 중복을 줄이면 가독성과 유지보수에 좋을 것 같습니다.

```suggestion
 const printComponent = (name, style, domElement = null) => {
  const [styledComponentName, styleString] = colorize(name, style);
  if (domElement) {
    console.log(`${line}▸ ${styledComponentName}`, styleString, domElement);
  } else {
    console.log(`${line}▸ ${styledComponentName}`, styleString);
  }
};
```